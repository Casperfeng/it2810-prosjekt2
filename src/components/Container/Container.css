.pane{
  display: flex;
  align-items: center;
  flex-direction: column;
}

.hBox{
  display: flex;
  flex-flow: row nowrap;
}
.vBox{
  display: flex;
  flex-flow: column nowrap;
}
.text{
  white-space: pre;
}