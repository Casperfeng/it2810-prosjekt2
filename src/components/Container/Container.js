import React from 'react'
import './Container.css';
import Tabs from './Tabs/Tabs.js'

function Container(props) {
  return (
    <div>
      <Tabs onTabIndexChanged={props.onTabIndexChanged} />
      <div className='container'>
        <svg id="svg" viewBox="0 0 100 100" dangerouslySetInnerHTML={{ __html: props.svg}}/>
        <div className='poem-container'>
          <p className='poem'>{props.poem}</p>
        </div>
        <audio className='audio' controls src={props.audio} type="audio/mpeg"/>
      </div>
    </div>
  );
}

export default Container;