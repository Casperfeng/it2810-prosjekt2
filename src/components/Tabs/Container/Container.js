import React from 'react'
import './Container.css';

function Container(props) {
  return (
    <div className='pane'>
      <div className='hBox'>
        <svg id="svg" width = "100" height = "100"
        dangerouslySetInnerHTML = {{ __html: props.svg}}/>
        <div className = 'vBox'>
          <p className = 'text'>{props.poem}</p>
          <audio controls src = {props.audio} type = "audio/mpeg"/>
        </div>
      </div>
    </div>
  );
}

export default Container;