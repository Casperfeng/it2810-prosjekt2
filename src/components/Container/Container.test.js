import React from 'react';
import Container from './Container';
import renderer from 'react-test-renderer';

it('renders Container correctly', () => {
  const tree = renderer
    .create(<Container />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});



