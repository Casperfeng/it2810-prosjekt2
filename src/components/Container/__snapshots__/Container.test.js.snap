// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Container correctly 1`] = `
<div>
  <div
    className="tabs-container"
  >
    <button
      className="tab "
      onClick={[Function]}
    >
      1
    </button>
    <button
      className="tab "
      onClick={[Function]}
    >
      2
    </button>
    <button
      className="tab "
      onClick={[Function]}
    >
      3
    </button>
    <button
      className="tab "
      onClick={[Function]}
    >
      4
    </button>
  </div>
  <div
    className="container"
  >
    <svg
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
      id="svg"
      viewBox="0 0 100 100"
    />
    <div
      className="poem-container"
    >
      <p
        className="poem"
      />
    </div>
    <audio
      className="audio"
      controls={true}
      type="audio/mpeg"
    />
  </div>
</div>
`;
